File: Python/60-把二叉树打印成多行.py
# -*- coding:utf-8 -*-
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None
class Solution:
    # 返回二维列表[[1,2],[4,5]]
    def Print(self, pRoot):
        # write code here
        if not pRoot:
            return []
        ret = []
        queue = [pRoot]
        while queue:
            t_queue = []
            t_ret = []
            while queue:
                node = queue.pop(0)
                t_ret.append(node.val)
                if node.left:
                    t_queue.append(node.left)
                if node.right:
                    t_queue.append(node.right)
            queue = t_queue
            ret.append(t_ret)
        return ret

File: C++/60-把二叉树打印成多行.cpp
/*
struct TreeNode {
    int val;
    struct TreeNode *left;
    struct TreeNode *right;
    TreeNode(int x) :
            val(x), left(NULL), right(NULL) {
    }
};
*/
class Solution
{
public:
    vector<vector<int>> Print(TreeNode *pRoot)
    {
        vector<vector<int>> ret;
        queue<TreeNode *> q;
        if (!pRoot)
            return ret;
        q.push(pRoot);
        while (!q.empty())
        {
            queue<TreeNode *> t_q;
            vector<int> t_ret;
            while (!q.empty())
            {
                TreeNode *node = q.front();
                q.pop();
                t_ret.push_back(node->val);
                if (node->left)
                    t_q.push(node->left);
                if (node->right)
                    t_q.push(node->right);
            }
            q = t_q;
            ret.push_back(t_ret);
        }
        return ret;
    }
};